source/event_loop.c

```diff
@@ -165,12 +165,16 @@ static struct aws_event_loop_group *s_event_loop_group_new(
 
     return el_group;
 
-on_error:
+on_error:;
+    /* cache the error code to prevent any potential side effects */
+    int cached_error_code = aws_last_error();
 
     aws_mem_release(alloc, usable_cpus);
     s_aws_event_loop_group_shutdown_sync(el_group);
```

Comment:
is one of those throwing some new error because it cant clean up correctly? should we fix that as well?

Reply:
There's code in here that loops through a priority list until it raises the AWS_ERROR_PRIORITY_QUEUE_EMPTY, indicating it's empty.

So it's not an error-as-fail, it's an error-code-as-flow-control